engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigrationState.java

```diff
@@ -202,4 +217,30 @@ public void migrateTemporaryVariables(
           temporaryVariableColumnFamily.deleteExisting(key);
         });
   }
+
+  @Override
+  public void migrateMultiInstanceChildActivatedCount(
+      final MutableElementInstanceState elementInstanceState) {
+    parentChildColumnFamily.forEach(
+        (key, value) -> {
+          final long parentElementInstanceKey = key.first().getValue();
+          if (parentElementInstanceKey != -1) {
+            final var elementInstance = elementInstanceState.getInstance(parentElementInstanceKey);
+            if (elementInstance != null
+                && elementInstance
+                    .getValue()
+                    .getBpmnElementType()
+                    .equals(BpmnElementType.MULTI_INSTANCE_BODY)) {
+              this.parentKey.wrapLong(parentElementInstanceKey);
+              parentChildColumnFamily.whileEqualPrefix(
+                  this.parentKey,
+                  (k, v) -> {
+                    elementInstance.incrementNumberOfElementInstances();
+                    elementInstanceState.updateInstance(elementInstance);
+                  });
+            }
+          }
+          parentChildColumnFamily.deleteExisting(key);
```

Comment:
❌ I don't think we should delete any of the entries of the parentChildColumnFamily. Updating the values like you do on lines 238 and 239 should be enough. Or am I missing something?